test/dogma/reporter/simple_test.exs

```diff
@@ -0,0 +1,152 @@
+defmodule Dogma.Reporter.SimpleTest do
+  use ShouldI
+
+  import ExUnit.CaptureIO
+
+  alias Dogma.Reporter.Simple
+  alias Dogma.Script
+  alias Dogma.Error
+
+  @one_file [ %Script{ errors: [] } ]
+  @many_files [ %Script{}, %Script{}, %Script{} ]
+
+  @no_errors %Script{ errors: [] }
+  @some_errors %Script{ errors: [%Error{}] }
+
+  @files_no_errors [
+    %Script{ path: "foo.ex",
+      errors: [%Error{ rule: Foo.BadCode, line: 44, message: "Awful." }]
+    },
+    %Script{} ]
+  @files_some_errors [
+    %Script{
+      path: "foo.ex",
+      errors: [%Error{ rule: Foo.BadCode,    line: 44, message: "Awful." }]
+    },
+    %Script{},
+    %Script{},
+    %Script{
+      path: "bar.ex",
+      errors: [
+        %Error{ rule: Foo.Confusing,  line: 2,  message: "Wtf?" },
+        %Error{ rule: Foo.UglyAsHell, line: 63, message: "Not ok." }
+      ]
+    },
+  ]
+
+  with ".handle_event" do
```

Comment:
Sounds like a very good idea to me. These functions are internal implementation details.